### PR TITLE
Reorganize layout of Hearing Details page

### LINKED ISSUE #13752: Hearing Details layout updates
### Description
Update the layout and information hierarchy of the Hearing Details page.

### Designs
Figma: https://www.figma.com/file/V87TZArfdurCGJiEjQ73ES/Virtual-Hearings?node-id=109%3A17309

### Acceptance criteria
  -  Hearing Details fields are re-arranged into the following sections separated by divider lines:
      - section with `VLJ`, `Hearing Coordinator`, `Hearing Room` fields
      - section with `Hearing Type` field
      -  section with Virtual Hearing Details (h3) heading; `VLJ Virtual Hearing Link`, `Veteran Email for Notifications`, and `POA/Representative Email for Notifications` fields; and the Email Notification History accordion/section
      -  section with `Waive 90 Day Evidence Hold` field
      -  section with `Notes` field
  -  The following Transcription Details headings are changed to h3:
      -  Transcription Problem
      -  Transcription Request

### Background/context/resources
This work has been broken out from Display more info about sent Virtual Hearings emails (#13370), which was based on usability testing with Hearing Coordinators (#12960)

### PR DESCRIPTION
**Pull request by ferristseng**
Resolves #13752, #13870

### Description

  - Remove `required` for veteran email field
  - Remove `DetailsSections`, and move transcription section to `DetailsInputs`
    - Required some small changes to use safe navigation in a variety of places
  - Use flex box instead of specifying width
  - Rearrange fields on form
  - Update `yarn run lint` to ignore deleted `js*` files

### Acceptance Criteria

*Hearing Details fields are re-arranged into the following sections separated by divider lines:*

- [x] section with VLJ, Hearing Coordinator, Hearing Room fields
- [x] section with Hearing Type field
- [x] section with Virtual Hearing Details (h3) heading; VLJ Virtual Hearing Link, Veteran Email for Notifications, and POA/Representative Email for Notifications fields; ~and the Email Notification History accordion/section~
- [x] section with Waive 90 Day Evidence Hold field
- [x] section with Notes field

*The following Transcription Details headings are changed to h3:*

- [x] Transcription Problem
- [x] Transcription Request

### User Facing Changes

---

localhost_3000_hearings_30ea3d0d-df9c-4eb0-a08a-2cd9ca8eaf9a_details

---

localhost_3000_hearings_3225545b-f562-4498-b247-1ec9504e3c3a_details

---

localhost_3000_hearings_404112_details

### CODE REVIEW

File: client/app/hearings/components/details/DetailsInputs.jsx [hunk 1]
```diff
@@ -68,135 +71,213 @@ EmailSection.propTypes = {
...
-      }
-      <EmailSection
-        hearing={hearing}
-        virtualHearing={virtualHearing}
-        isVirtual={isVirtual}
-        wasVirtual={wasVirtual}
-        readOnly={readOnly}
-        updateVirtualHearing={updateVirtualHearing}
-      />
+    <React.Fragment>
```
[the remark this answers is not in the document]
Comment (ferristseng) [reply client/app/hearings/components/details/DetailsInputs.jsx:203]: This came up during pairing too, but I prefer `<>` personally. Not sure what is the standard in the React community.

**Comment by ferristseng** [comment]
> A non-blocking nit: because the new labels for the email fields are so long, they start wrapping when the browser window width is < 1302px (mac/chrome) and it looks messy, because there's plenty of space to the right and it pushes the fields out of horizontal alignment:

Not super sure if flexbox can handle this easily, but I can experiment with it.

**Comment by ferristseng** [comment]
Fixed @tomas-nava!

Screen Recording 2020-04-13 at 4 34 38 PM

**Comment by nicholer** [comment]
Overall this looks great 🤩

One small comment about the h3 spacing: it looks like the Virtual Hearing Details heading has less bottom spacing than the Transcription headings. I prefer the amount of separation in the Transcription headings; is it possible to have Virtual Hearing Details match?

Screen Shot 2020-04-13 at 4 46 53 PM 
Screen Shot 2020-04-13 at 4 47 10 PM

**Comment by ferristseng** [comment]
> One small comment about the h3 spacing: it looks like the Virtual Hearing Details heading has less bottom spacing than the Transcription headings. I prefer the amount of separation in the Transcription headings; is it possible to have Virtual Hearing Details match?

Yes, I think that should be possible @nicholer . Let me try to find where that style gets set!

**Comment by ferristseng** [comment]
@nicholer Spacing should be consistent now:

Screen Shot 2020-04-13 at 5 11 33 PM

**Pull request merged**